contracts/ReversibleICO.sol

```diff
@@ -253,26 +253,26 @@ contract ReversibleICO is IERC777Recipient {
 
         // Setup stage 1 to n: The buy phase stages
         // Each new stage starts after the previous phase's endBlock
-        uint256 newStageStartBlock = stage0.endBlock + 1;
+        uint256 newStageEndBlock = stage0.endBlock;
 
         for (uint8 i = 1; i <= _stageCount; i++) {
 
             Stage storage stageN = stages[stageCount];
             // stageCount = n
-            stageN.startBlock = newStageStartBlock;
-            stageN.endBlock = newStageStartBlock + _stageBlockCount;
+            stageN.startBlock = newStageEndBlock + 1;
```

Comment:
Looks like you're reverting everything you did in your first commits and make it like before.

Reply:
I would rename newStageEndBlock to previousStageEndBlock, as that marks the variable clearly

Reply:
agreed, done!